common/psbt_open.c

```diff
@@ -408,13 +386,13 @@ bool psbt_has_required_fields(struct wally_psbt *psbt)
 			return false;
 
 		/* If is P2SH, redeemscript must be present */
-		assert(psbt->tx->inputs[i].index < input->utxo->num_outputs);
+		assert(psbt->inputs[i].index < input->utxo->num_outputs);
 		const u8 *outscript =
 			wally_tx_output_get_script(tmpctx,
-				&input->utxo->outputs[psbt->tx->inputs[i].index]);
-		if (is_p2sh(outscript, NULL) && input->redeem_script_len == 0)
+				&input->utxo->outputs[psbt->inputs[i].index]);
+		const struct wally_map_item *redeem_script = wally_map_get_integer(&psbt->inputs[i].psbt_fields, /* PSBT_IN_REDEEM_SCRIPT */ 0x04);
```

Comment:
Yeah, this is also probably initializer-abuse.

Reply:
done